pkg/ruler/frontend_client_pool.go

```diff
@@ -0,0 +1,83 @@
+package ruler
+
+import (
+	"time"
+
+	"github.com/go-kit/log"
+	otgrpc "github.com/opentracing-contrib/go-grpc"
+	"github.com/opentracing/opentracing-go"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/weaveworks/common/httpgrpc"
+	"github.com/weaveworks/common/middleware"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/health/grpc_health_v1"
+
+	"github.com/cortexproject/cortex/pkg/ring/client"
+	cortexmiddleware "github.com/cortexproject/cortex/pkg/util/middleware"
+)
+
+type frontendPool struct {
+	timeout              time.Duration
+	prometheusHTTPPrefix string
+	grpcConfig           ClientConfig
```

Comment:
I wonder if we should use `grpcclient.Config` instead? 
`ClientConfig` seems used for Ruler to talk to each other only